src/core/User/service/UserProfileService.ts

```diff
@@ -129,4 +129,11 @@ export class UserProfileService {
 
     return currentRoleIndex > targetRoleIndex
   }
+
+  async deleteUsersByCompany(companyDomain: string): Promise<void> {
```

Comment:
Se dovesse esserci un errore ad un certo punto del ciclo for la cancellazione degli utenti verrebbe interrotta lasciando il DB in stato inconsistente. Andrebbe fatta un'unica transazione